server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -171,8 +180,47 @@ public Stream<Reference> getReferences() {
   }
 
   @Override
-  public Set<TableId> getTableIDs() {
-    return context.getTableIdToNameMap().keySet();
+  public Map<TableId,TableState> getTableIDs() throws InterruptedException {
+    final String tablesPath = context.getZooKeeperRoot() + Constants.ZTABLES;
+    final ZooReader zr = context.getZooReader();
+    int retries = 1;
+    UncheckedIOException ioe = null;
+    while (retries <= 10) {
+      try {
+        zr.sync(tablesPath);
+        final Map<TableId,TableState> tids = new HashMap<>();
+        for (String table : zr.getChildren(tablesPath)) {
+          TableId tableId = TableId.of(table);
+          TableState tableState = null;
+          String statePath = context.getZooKeeperRoot() + Constants.ZTABLES + "/"
+              + tableId.canonical() + Constants.ZTABLE_STATE;
+          try {
+            byte[] state = zr.getData(statePath, null, null);
+            if (state == null) {
+              tableState = TableState.UNKNOWN;
+            } else {
+              tableState = TableState.valueOf(new String(state, UTF_8));
+            }
+          } catch (KeeperException e) {
+            if (e.code() == Code.NONODE) {
+              tableState = TableState.UNKNOWN;
+            }
+            throw e;
```

Comment:
Should this be if/else?

```suggestion
            if (e.code() == Code.NONODE) {
              tableState = TableState.UNKNOWN;
            } else {
               throw e;
             }
```

May not matter if we change to use NoNodeException that I commented on in other thread.

Reply:
Yeah, I think I'm rushing.....